go/fix-chain/urlcache.go

```diff
@@ -0,0 +1,63 @@
+package fix_chain
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"time"
+)
+
+type URLCache struct {
+	cache map[string][]byte
+	// counters may not be totally accurate due to non-atomicity
+	hit       uint
+	miss      uint
+	errors    uint
+	badstatus uint
+	readfail  uint
+}
+
+func (u *URLCache) getURL(url string) ([]byte, error) {
+	r, ok := u.cache[url]
+	if ok {
+		u.hit++
+		return r, nil
+	}
+	c, err := http.Get(url)
+	// FIXME: cache errors
+	if err != nil {
+		u.errors++
+		return nil, err
+	}
+	defer c.Body.Close()
+	if c.StatusCode != 200 {
+		u.badstatus++
+		return nil, errors.New(fmt.Sprintf("can't deal with status %d", c.StatusCode))
+	}
+	r, err = ioutil.ReadAll(c.Body)
+	if err != nil {
+		u.readfail++
+		return nil, err
+	}
+	u.miss++
+	u.cache[url] = r
+	return r, nil
+}
+
+func NewURLCache() *URLCache {
+	u := &URLCache{cache: make(map[string][]byte)}
+
+	t := time.NewTicker(time.Second)
+	go func() {
+		for _ = range t.C {
+			log.Printf("cache: %d hits, %d misses, %d errors, " +
```

Comment:
Same thing here, embrace the Go-ness, just stick it on the same line (or, at least run `gofmt` and let it fix it up).